lib/clickhousex/http_client.ex

```diff
@@ -97,13 +97,18 @@ defmodule Clickhousex.HTTPClient do
     else
       {:error, conn, error, _messages} ->
         {:error, conn, error}
+
+      error ->
+        error
     end
   end
 
   defp decode_response(conn, %Query{type: :select}, %Response{} = response) do
     case Response.decode(response) do
       {:ok, %{column_names: columns, rows: rows}} -> {:ok, conn, {:selected, columns, rows}}
-      {:error, error} -> {:error, conn, error.reason}
```

Comment:
Not all errors come in a map with `:reason` set. Actually, I haven't seen this pattern used anywhere at all.